docs/developing-fuzzbench/adding_a_new_benchmark.md

```diff
@@ -71,86 +61,148 @@ Add the files in the benchmark directory to git (and then commit them):
-## Standard benchmarks: Create benchmark files
+```bash
+cd benchmarks
+export BENCHMARK_NAME=<your_benchmark_name>
+mkdir $BENCHMARK_NAME
+```
 
-### fuzz_target.cc
+### Defining a fuzz target
 
-This file defines the entry point for fuzzing. It should define a
-`LLVMFuzzerTestOneInput` function that accepts an array of bytes and does
-something interesting with these bytes using the program API under test.
+Benchmarks in OSS-Fuzz consist of open source code and [a libFuzzer compatible
+entrypoint](https://llvm.org/docs/LibFuzzer.html#fuzz-target) into the targeted
+code that fuzzers such as AFL, libFuzzer and honggfuzz send data to fuzz.
+This section describes how to create a file that defines this entrypoint.
+This file  should define a
+`LLVMFuzzerTestOneInput` function that accepts an array of bytes and the
+length of this array. This function should then pass those bytes to an API
+in the project/program that we want to fuzz.
 
 ```c
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
   DoSomethingInterestingWithMyAPI(Data, Size);
   return 0;  // Non-zero return values are reserved for future use.
 }
 ```
+For example, if the project we are fuzzing is a JSON parsing library, our
+`LLVMFuzzerTestOneInput` could pass the data from the fuzzer to a function in
+the library that parses JSON.
 
 Example: [libxml2](https://github.com/google/fuzzbench/blob/master/benchmarks/libxml2-v2.9.2/target.cc).
 
 ### build.sh
 
-This file builds the fuzz target for your benchmark, taking into account
-the compiler options - `CC`, `CXX`, `CFLAGS`, `CXXFLAGS` and the fuzzer library
+This file builds the fuzz target for your benchmark. It should use the
+environment variables `CC`, `CXX`, `CFLAGS`, `CXXFLAGS` `$OUT` and the fuzzer
+library.
 `FUZZER_LIB` ([explanation]({{ site.baseurl }}/getting-started/adding-a-new-fuzzer/#what-is-fuzzer_lib))
-as input.
+for this. These environment variables will be defined when `build.sh` runs
+By linking your fuzz target with the `FUZZER_LIB` and the project under test,
+you will produce a binary that can be fuzzed by the FuzzBench fuzzers. This is
+called the fuzz target binary. `FUZZER_LIB` is specific to each fuzzer, its
+primary purpose is taking data from the fuzzer and passing it to your
+benchmark's `LLVMFuzzerTestOneInput` function.
 
 Once the build is finished, copy the fuzz target binary, any library
-dependencies and the seeds directory into the output directory (`$OUT`).
+dependencies (and optionally the seeds directory and the dictionary)
+into the output directory (`$OUT`).
 **NOTE**: Only build artifacts added in `$OUT` directory are available when
 running the fuzzer. You should not have any dependencies outside of `$OUT`.
 
-
 ```bash
 #!/bin/bash -ex
-. $(dirname $0)/../common.sh
 
-build_project() {
-  rm -rf BUILD
-  cp -rf SRC BUILD
-  (cd BUILD && ./autogen.sh && ./configure && make -j $JOBS)
-}
-get_git_revision <path-to-git-repo> <git-hash> SRC
-build_project
+rm -rf BUILD
```

Comment:
Do we need to delete BUILD at all. Can we remove this FTP hack.

Reply:
Removed it.